src/main/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -199,8 +200,8 @@ private SAMRecord createSamRecord(final ReadData readData, final String readName
             sam.setAttribute(ReservedTagConstants.XN, 1);
         }
 
-        if (this.readGroupId != null) {
-            sam.setAttribute(SAMTag.RG.name(), readGroupId);
+        if (this.laneToReadGroup.containsKey(lane)) {
+            sam.setAttribute(SAMTag.RG.name(), laneToReadGroup.get(lane));
```

Comment:
```suggestion
        final String rgId = this.laneToReadGroup.get(lane);
        if (rgId != null) sam.setAttribute(SAMTag.RG.name(), rgId);
```
Avoid hitting the map twice.